src/core/localStore.js

```diff
@@ -0,0 +1,76 @@
+/* @flow */
+import defaultLocalForage from 'localforage';
+
+import log from 'core/logger';
+
+export function configureLocalForage(
+  {
+    localForage = defaultLocalForage,
+  }: {|
+    localForage: typeof defaultLocalForage,
+  |} = {}
+) {
+  localForage.config({
+    name: 'addons-frontend',
+    version: '1.0',
+    storeName: 'core.localStore',
+  });
+}
+
+type LocalStoreOptions = {|
+  localForage: typeof defaultLocalForage,
+|};
+
+export class LocalStore {
+  id: string;
+  localForage: typeof defaultLocalForage;
+
+  constructor(
+    id: string,
+    { localForage = defaultLocalForage }: LocalStoreOptions = {},
+  ) {
+    this.id = id;
+    this.localForage = localForage;
+    configureLocalForage({ localForage });
+  }
+
+  getData(): Promise<Object | null> {
```

Comment:
Ah okay I see how this works now, we limit our storage engine to only use one key and put the data in there.

Maybe I'm being pedantic but I don't think adding `Data` to all of these methods helps with anything. I think we could remove them and it would make as much sense.

I might just be allergic to the word "data" in code though. Maybe I should just take some anti-allergy meds and be quiet 😉